lib/active_record/cursor_paginator.rb

```diff
@@ -44,10 +47,11 @@ def extract_order_fields(relation)
         case o
         when Arel::Attribute # .order(arel_table[:id])
           { o.name => :asc }
-        when Arel::Nodes::Ascending # .order(id: :asc), .order(:id)
-          { o.expr.name => :asc }
-        when Arel::Nodes::Descending # .order(id: :desc)
-          { o.expr.name => :desc }
+        when Arel::Nodes::Ascending, # .order(id: :asc), .order(:id)
+             Arel::Nodes::Descending # .order(id: :desc)
+          key = o.expr.is_a?(Arel::Attributes::Attribute) ? o.expr.name : trim_quote(o.expr)
+          dir = o.is_a?(Arel::Nodes::Descending) ? :desc : :asc
```

Comment:
```suggestion
          dir = o.direction
```

でいけそうですかね。
https://www.rubydoc.info/gems/arel/Arel/Nodes/Descending#direction-instance_method